### ISSUE TITLE
Missing text on GUI

### ISSUE THREAD

**Issue by auroraanna** [PolyMeilex/Neothesia#77]
On master (d724865059e876f14a3b92ba18a1c5dbe3485131), the text on the GUI is missing, e.g. all the buttons, select menus but now the hints for what key plays what note. Therefore I guess, assuming that Neothesia renders the note hints, not iced, that this is an issue with iced.

The [previous issue about this](#11) seems to have fixed this by switching to DX12. I am on Linux so this application should only be able to run on Vulkan on my system.

Screenshots of what it looks like:
![image](https://github.com/PolyMeilex/Neothesia/assets/81317317/25eddc20-cc26-43e9-84c1-84e516f31556)
![image](https://github.com/PolyMeilex/Neothesia/assets/81317317/01a33444-b8b9-42b8-ad6b-9a2b60573b5b)

This doesn't occur on 0.0.16 and for 1.0.0, I can't get it building in nixpkgs (https://github.com/NixOS/nixpkgs/pull/256882).

**Comment by PolyMeilex** [comment]
Yep, text rendered while playing the song is rendered by Neothesia directly, in menu it's done by iced

**Comment by PolyMeilex** [comment]
You could try with opengl

**Comment by auroraanna** [comment]
> You could try with opengl

So, I figured you can set the `WGPU_BACKEND` environment variable but it launches successfully on nothing but `gl`. So it must've been running on OpenGL this whole time. I thought vulkan would work but no.

```console
$ WGPU_BACKEND="vulkan" ./result/bin/neothesia
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: AdapterRequest', neothesia/src/main.rs:243:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: queue 0x55b0209b1650 destroyed while proxies still attached:
  xdg_wm_base@20 still attached
  wl_output@14 still attached
  wl_seat@13 still attached
  xdg_activation_v1@12 still attached
  wp_viewporter@11 still attached
  zwp_text_input_manager_v3@10 still attached
  zwp_pointer_constraints_v1@9 still attached
  zwp_relative_pointer_manager_v1@8 still attached
  zxdg_decoration_manager_v1@7 still attached
  wl_subcompositor@6 still attached
  wl_compositor@5 still attached
  wl_shm@4 still attached
  wl_registry@2 still attached
```

**Comment by PolyMeilex** [comment]
Could you try with "RUST_LOG=debug" env

**Comment by auroraanna** [comment]
> Could you try with "RUST_LOG=debug" env

With that env variable I found out the problem with running on vulkan was that it didn't have libvulkan.so.1 in it's path which it has now.

So, it runs on vulkan but the text is still missing. However,  the Neothesia logo shows now:
![image](https://github.com/PolyMeilex/Neothesia/assets/81317317/912c9db8-3cb4-4756-8a27-5988e474877a)

**Comment by auroraanna** [comment]
Oh, it was logging that it failed to find a bunch of fonts which I have all installed. Recompiling with fontconfig in runtime path.

**Comment by auroraanna** [comment]
Hmm, fontconfig in it's runtime PATH doesn't help. For vulkan-loader with libvulkan.so.1 I had to explicitly put it's lib folder into `LD_LIBRARY_PATH` though. Maybe fontconfig is not accessible to it?

**Comment by auroraanna** [comment]
> The database can try to load system fonts. Currently, this is implemented by scanning predefined directories. The library does not interact with the system API.
>
> --<https://crates.io/crates/fontdb>

I looked in the dependency graph and found fontdb. Assuming this crate is used for using system fonts, scanning predefined directories doesn't work on my Linux distro (NixOS). That must be why adding fontconfig to the runtime PATH didn't help.

I will try to put it in a FHS environment then so the predefined directories will work.

**Comment by auroraanna** [comment]
So, I was told putting it in an FHS env is not ideal. Patching the source code of the fontdb dependency would be better, but complex. Another option is Neothesia switching to using fontconfig. You can't expect fonts to be in predefined paths. Not every systen works the same way.

**Comment by PolyMeilex** [comment]
I wonder if it would be possible to hardcore the iced font the same way I do for Neothesia, for Neothesia I just prebundle the hardcoded font and that's why it works, iced on the other hand is "smarter" and tries to use system's font.

Could you try one of iced demos/examples? So we can at least report this upstream before I strip the font detection part out (if it manifests upstream that is)

**Comment by auroraanna** [comment]
> So we can at least report this upstream

Upstream in iced? Shouldn't this be reported in cosmic-text or fontdb itself? (fontdb is a direct dependency of cosmic-text)

**Comment by auroraanna** [comment]
Hmm, there's already issues about this:
- <https://github.com/RazrFalcon/fontdb/issues/56>
- <https://github.com/pop-os/cosmic-text/issues/167>
